File: SalesOrder/SalesOrder.Model.VB/Customer.vb
Namespace SalesOrder.Model
    Public Class Customer

        Public Overridable Property Id As Integer

        Public Overridable Property Name As String

        Public Overridable Property Email As String

        Public Overridable Property Orders As ICollection(Of Order) = New List(Of Order)

        Public Overridable Property Addresses As ICollection(Of Address) = New List(Of Address)

    End Class
End Namespace

File: SalesOrder/SalesOrder.Model/Customer.cs
using System.Collections.Generic;

namespace SalesOrder.Model
{
    public class Customer
    {
        public virtual int Id { get; set; }

        public virtual string Name { get; set; }

        public virtual string Email { get; set; }

        public virtual ICollection<Order> Orders { get; set; } = new List<Order>();     

        public virtual ICollection<Address> Addresses { get; set; } = new List<Address>();
    }
}